### ISSUE TITLE
RedundantSafeNavigation does not consider constants may be nil.

### ISSUE THREAD

**Issue by ioquatix** [rubocop/rubocop#12271]
## Expected behavior

```
FOO = nil

self::FOO&.dig(...)
      ^^^^^
[Correctable] Lint/RedundantSafeNavigation: Redundant safe navigation detected.
```

## Actual behavior

I don't believe we should flag constants as not needing safe navigation.

## Steps to reproduce the problem

This is extremely important! Providing us with a reliable way to reproduce
a problem will expedite its solution.

## RuboCop version

```
1.57.0 (using Parser 3.2.2.4, rubocop-ast 1.29.0, running on ruby 3.2.2) [x86_64-linux]
  - rubocop-rspec 2.24.1

```

## Potential Fix

Consider deleting this line? https://github.com/rubocop/rubocop/blob/128618c7a8e3336bbe9ff8f800a73ff04dfc7a81/lib/rubocop/cop/lint/redundant_safe_navigation.rb#L76

**Comment by ioquatix** [comment]
Thanks everyone for your quick turn around!
